backend/contributions/views.py

```
@@ -3516,8 +3377,14 @@ def get_queryset(self):
 
         # Filter by contribution type if specified
-        contribution_type = self.request.query_params.get('contribution_type', None)
-        if contribution_type:
+        contribution_type = self.request.query_params.get('contribution_type')
+        if contribution_type not in (None, ''):
+            try:
+                contribution_type = int(contribution_type)
+            except (TypeError, ValueError):
+                raise serializers.ValidationError({
+                    'contribution_type': 'Must be a valid contribution type id.'
+                })
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)